### PR TITLE
treewide: replace name with pname

### PR DESCRIPTION
**Pull request by jopejoe1**
## Description of changes
If name is used instead of pname, our [search page](https://search.nixos.org/packages) fails to display the version number for the package.

The build failures are not related to this pr and already happened on master.

Note: there are still more of those flying around.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Result of `nixpkgs-review pr 315183` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>bombsquad</li>
    <li>pacup</li>
    <li>pacup.dist</li>
  </ul>
</details>
<details>
  <summary>26 packages built:</summary>
  <ul>
    <li>c64-debugger</li>
    <li>cano</li>
    <li>disko</li>
    <li>dorion</li>
    <li>holochain-launcher</li>
    <li>jailer</li>
    <li>jtdx</li>
    <li>kamilalisp</li>
    <li>kdiskmark</li>
    <li>lcrq</li>
    <li>lcsync</li>
    <li>libgourou</li>
    <li>liblapin</li>
    <li>liblapin.dev</li>
    <li>librecast</li>
    <li>lssecret</li>
    <li>menulibre</li>
    <li>menulibre.dist</li>
    <li>patcher9x</li>
    <li>physac</li>
    <li>pixel-code</li>
    <li>polychromatic</li>
    <li>python311Packages.raylib-python-cffi</li>
    <li>python311Packages.raylib-python-cffi.dist</li>
    <li>python312Packages.raylib-python-cffi</li>
    <li>python312Packages.raylib-python-cffi.dist</li>
  </ul>
</details>

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 315183` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>bombsquad</li>
    <li>pacup</li>
    <li>pacup.dist</li>
  </ul>
</details>
<details>
  <summary>26 packages built:</summary>
  <ul>
    <li>c64-debugger</li>
    <li>cano</li>
    <li>disko</li>
    <li>dorion</li>
    <li>holochain-launcher</li>
    <li>jailer</li>
    <li>jtdx</li>
    <li>kamilalisp</li>
    <li>kdiskmark</li>
    <li>lcrq</li>
    <li>lcsync</li>
    <li>libgourou</li>
    <li>liblapin</li>
    <li>liblapin.dev</li>
    <li>librecast</li>
    <li>lssecret</li>
    <li>menulibre</li>
    <li>menulibre.dist</li>
    <li>patcher9x</li>
    <li>physac</li>
    <li>pixel-code</li>
    <li>polychromatic</li>
    <li>python311Packages.raylib-python-cffi</li>
    <li>python311Packages.raylib-python-cffi.dist</li>
    <li>python312Packages.raylib-python-cffi</li>
    <li>python312Packages.raylib-python-cffi.dist</li>
  </ul>
</details>

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 315183` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>c64-debugger</li>
    <li>pacup</li>
    <li>pacup.dist</li>
  </ul>
</details>
<details>
  <summary>26 packages built:</summary>
  <ul>
    <li>bombsquad</li>
    <li>cano</li>
    <li>disko</li>
    <li>dorion</li>
    <li>holochain-launcher</li>
    <li>jailer</li>
    <li>jtdx</li>
    <li>kamilalisp</li>
    <li>kdiskmark</li>
    <li>lcrq</li>
    <li>lcsync</li>
    <li>libgourou</li>
    <li>liblapin</li>
    <li>liblapin.dev</li>
    <li>librecast</li>
    <li>lssecret</li>
    <li>menulibre</li>
    <li>menulibre.dist</li>
    <li>patcher9x</li>
    <li>physac</li>
    <li>pixel-code</li>
    <li>polychromatic</li>
    <li>python311Packages.raylib-python-cffi</li>
    <li>python311Packages.raylib-python-cffi.dist</li>
    <li>python312Packages.raylib-python-cffi</li>
    <li>python312Packages.raylib-python-cffi.dist</li>
  </ul>
</details>

**Review by l0b0 — APPROVED** [review]
Code LGTM; haven't tested any of the builds.

File: pkgs/misc/lssecret/default.nix [hunk 1]
```diff
@@ -6,12 +6,12 @@
...
 
 stdenv.mkDerivation rec {
-  name = "lssecret";
+  pname = "lssecret";
   version = "unstable-2022-12-02";
 
   src = fetchFromGitLab {
     owner = "GrantMoyer";
-    repo = name;
+    repo = pname;
```
Comment (drupol) [inline pkgs/misc/lssecret/default.nix:14]: I would not do that. I would replace `fetchFromGitLab.pname` with a literal. 
```suggestion
    repo = "lssecret";
```

File: pkgs/tools/package-management/pacup/default.nix [hunk 2]
```diff
@@ -4,13 +4,13 @@
...
 python3.pkgs.buildPythonApplication rec {
-  name = "pacup";
+  pname = "pacup";
   version = "2.0.0";
   format = "pyproject";
 
   src = fetchFromGitHub {
     owner = "pacstall";
-    repo = name;
+    repo = pname;
```
Comment (drupol) [inline pkgs/tools/package-management/pacup/default.nix:13]:
```suggestion
    repo = "pacup";
```

File: pkgs/development/libraries/libgourou/default.nix [hunk 3]
```diff
@@ -10,11 +10,11 @@
 }:
 
 stdenv.mkDerivation rec {
-  name = "libgourou";
+  pname = "libgourou";
   version = "0.8.2";
 
   src = fetchzip {
-    url = "https://indefero.soutade.fr/p/${name}/source/download/v${version}/";
+    url = "https://indefero.soutade.fr/p/${pname}/source/download/v${version}/";
```
Comment (drupol) [inline pkgs/development/libraries/libgourou/default.nix:17]:
```suggestion
    url = "https://indefero.soutade.fr/p/libgourou/source/download/v${version}/";
```

File: pkgs/development/interpreters/kamilalisp/default.nix [hunk 4]
```diff
@@ -20,9 +20,9 @@ stdenv.mkDerivation rec {
 
   installPhase = ''
     mkdir -pv $out/share/java $out/bin
-    cp ${src} $out/share/java/${name}-${version}.jar
+    cp ${src} $out/share/java/${pname}-${version}.jar
```
Comment (drupol) [inline pkgs/development/interpreters/kamilalisp/default.nix:23]:
```suggestion
    cp ${src} $out/share/java/kamilalisp-${version}.jar
```

File: pkgs/development/interpreters/kamilalisp/default.nix [hunk 5]
```diff
@@ -20,9 +20,9 @@ stdenv.mkDerivation rec {
...
     makeWrapper ${jre}/bin/java $out/bin/kamilalisp \
-      --add-flags "-jar $out/share/java/${name}-${version}.jar" \
+      --add-flags "-jar $out/share/java/${pname}-${version}.jar" \
```
Comment (drupol) [inline pkgs/development/interpreters/kamilalisp/default.nix:25]:
```suggestion
      --add-flags "-jar $out/share/java/kamilalisp-${version}.jar" \
```

**Review by drupol — CHANGES_REQUESTED** [review]
Thanks for doing this PR, however I don't agree with reusing `pname` here and there. If you modify a line, make sure it is valid and following best practices.

I added code suggestion comments to explain my view on this.

**Comment by jopejoe1** [comment]
Updated to not reuse pname.

**Review by drupol — APPROVED** [review]
Thank you !

**Pull request merged**